### PR TITLE
verilator: 4.210 -> 4.218

### PR DESCRIPTION
**Pull request by nekronos**
###### Motivation for this change
Version bump

###### Things done
Change to fetch source from github as the source for newer versions are not published at the same location anymore.  Also add configure/build/install-phases as verilator uses autoconf (autoreconfHook is not supported by verilator).

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/science/electronics/verilator/default.nix [hunk 1]
```diff
@@ -1,25 +1,40 @@
...
   pname = "verilator";
-  version = "4.210";
+  version = "4.218";
 
-  src = fetchurl {
-    url = "https://www.veripool.org/ftp/${pname}-${version}.tgz";
-    sha256 = "sha256-KoIfJeV2aITnwiB2eQgQo4ZyXfMe6erFiGKXezR+IBg=";
+  src = fetchFromGitHub {
+    owner = "${pname}";
+    repo = "${pname}";
```
Comment (SuperSandro2000) [inline pkgs/applications/science/electronics/verilator/default.nix:11]:
```suggestion
    owner = pname;
    repo = pname;
```

File: pkgs/applications/science/electronics/verilator/default.nix [hunk 2]
```diff
@@ -1,25 +1,40 @@
...
+  '';
+
+  buildPhase = ''
+    make -j $NIX_BUILD_CORES
+  '';
+
+  installPhase = ''
+    make install
+  '';
+
```
Comment (SuperSandro2000) [inline pkgs/applications/science/electronics/verilator/default.nix:37]:
```suggestion
```
All of this should happen automatically.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by nekronos** [comment]
@SuperSandro2000 the package does not build with this change https://github.com/NixOS/nixpkgs/pull/157122/commits/71cbfd1a727fc77d6665ab8c753ec0b510308b6e

**Comment by SuperSandro2000** [comment]
> @SuperSandro2000 the package does not build with this change [71cbfd1](https://github.com/NixOS/nixpkgs/commit/71cbfd1a727fc77d6665ab8c753ec0b510308b6e)

I need some more details than that to help you.

**Comment by nekronos** [comment]
> > @SuperSandro2000 the package does not build with this change [71cbfd1](https://github.com/NixOS/nixpkgs/commit/71cbfd1a727fc77d6665ab8c753ec0b510308b6e)
> 
> I need some more details than that to help you.

Looked into this in more detail. `autoconf` is not run automatically, so I have changed the code to run `autoconf` as a `preConfigurePhase`. Otherwise the correct configure and make calls happens automatically as you said and the package builds 👍🏻

File: pkgs/applications/science/electronics/verilator/default.nix [hunk 3]
```diff
@@ -1,25 +1,31 @@
...
   buildInputs = [ perl ];
-  nativeBuildInputs = [ flex bison python3 ];
+  nativeBuildInputs = [ flex bison python3 autoconf ];
 
   # these tests need some interpreter paths patched early on...
   # see https://github.com/NixOS/nix/issues/1205
   doCheck = false;
   checkTarget = "test";
 
+  preConfigurePhases = ''
```
Comment (SuperSandro2000) [inline pkgs/applications/science/electronics/verilator/default.nix:25]:
```suggestion
  preConfigure = ''
```

**Pull request merged**